### ISSUE TITLE
Local install failed on Mac when you already have ~/.ngrok as a file

### ISSUE THREAD

**Issue by trickpeeraze** [bubenshchykov/ngrok#236]
Check out the logic here

https://github.com/bubenshchykov/ngrok/blob/fd9e6d14080ad6c80d94f40ba09ab5e8711a27a4/download.js#L64-L74

There's a possible error in the case that `dir` is a file. It exists but not a folder. This will raise the exception and goes to the catch block. So, this is my point.

```js
dir = path.join(__dirname, "..", "bin");
```

creates a wrong path related to the current directory so that the script cannot download the ngrok's client correctly.
And it means the entire installation will fail.

Should we fix this somehow?

Thank you.

**Comment by philnash** [comment]
Is that an error that has happened to you? And if so, what were the circumstances?

The line you call out refers to the `bin` directory within the module file structure, which we know to be a directory because the structure is under our control. That is the fallback option because it should always work.

**Comment by trickpeeraze** [comment]
I first think that it should always work as you said but I'm not sure why it relates to the write permission. I got this kind of error.

```
ngrok - downloading binary https://bin.equinox.io/c/4VmDzA7iaHb/ngrok-stable-darwin-amd64.zip
ngrok - downloading progress: ngrok - error storing binary to local file { [Error: ENOENT: no such file or directory, open '/path/to/my/project/node_modules/bin/aHR0cHM6Ly9iaW4uZXF1aW5veC5pby9jLzRWbUR6QTdpYUhiL25ncm9rLXN0YWJsZS1kYXJ3aW4tYW1kNjQuemlw.zip']
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '/path/to/my/project/node_modules/bin/aHR0cHM6Ly9iaW4uZXF1aW5veC5pby9jLzRWbUR6QTdpYUhiL25ncm9rLXN0YWJsZS1kYXJ3aW4tYW1kNjQuemlw.zip' }
```

For my case, I have to use _.ngrok_ as a config file because of my company's internal tool.

**Comment by philnash** [comment]
Ah, ok, I thought you were suggesting that there might be an issue, not that you have indeed experienced an issue and need something to be fixed.

Can I confirm, `~/.ngrok` is a file on your machine? And that is triggering the issue in the download?

**Comment by trickpeeraze** [comment]
Yes, it's a file on my machine and it triggers the issue in the download. And yes I'm both suggesting and experiencing the issue 😅.

**Comment by philnash** [comment]
Yep, I see the issue now. Should be fixed by #237.

**Comment by philnash** [comment]
This has been released as version 4.1.0. Please let me know if it fixes your issues.

**Comment by trickpeeraze** [comment]
It's solved now. Thank you!
